Remove StarknetEthProxyCheck (#1292)
software-mansion/starknet.py: docs/migration_guide.rst

Migration guide
===============

**********************
0.20.0 Migration guide
**********************

0.20.0 Breaking Changes
-----------------------

> - :class:`StarknetEthProxyCheck` has been removed from the Proxy checks

Reviewer: nit: Can be changed in other PR
```suggestion
1. :class:`StarknetEthProxyCheck` has been removed from the Proxy checks
```

Migration guide
===============

**********************
0.20.0 Migration guide
**********************

> 0.20.0 Deprecations

Reviewer: I would consider categorizing this under `Breaking Changes`. Technically, it's mostly a breaking change on the Starknet side, yet it definitely doesn't classify as a deprecation.

-------------------

> .. currentmodule::  ~starknet_py.proxy.proxy_check

Reviewer: nit: It can be removed

> - :class:`StarknetEthProxyCheck` has been removed, because StarkGate ETH contract isn't proxy anymore

Reviewer: ```suggestion
- :class:`StarknetEthProxyCheck` has been removed from the Proxy checks
```
